### ISSUE TITLE
Error exporting data from OpenSGID to a new feature class in ArcGIS Desktop 10.8.1

### ISSUE THREAD

**Issue by gis4ag** [agrc/open-sgid#22]
About half the times I try to download data from OpenSGID, I get the following error:

![ExportError](https://user-images.githubusercontent.com/14881343/105750719-54ab0700-5f02-11eb-9ee2-bb817bde8d19.png)

It always works eventually, although it can take several tries. Any thoughts on what might cause this? In the example above, UtahParcels is the feature class I'm trying to extract data to.

Thanks for your help!

**Comment by steveoh** [comment]
would you provide a screen shot of the layers data source please. thanks!

**Comment by gis4ag** [comment]
Steve -

1.  ![LayersDataSource](https://user-images.githubusercontent.com/14881343/105753195-86719d00-5f05-11eb-8671-3c11a7f6d09a.png)

2. version is 10.8.1.14362
3.

**Comment by gis4ag** [comment]
oops...

3.  Intersect with parcel layer to get a select set of records, export to a new empty feature class.

**Comment by steveoh** [comment]
@gregbunce can we reproduce this?

**Comment by jacobdadams** [comment]
I have been able to reproduce this in at least one scenario. I will edit this as I dig more. It appears to be related to the spatial index (https://support.esri.com/en/technical-article/000015090) and how this works with how the Open SGID's source layer is added to the map (specifically, whether you let it calculate the extent or just use the spatial reference's extent).

**Comment by jacobdadams** [comment]
Ok, these errors are really random, I've not been able to pin down a workflow that is 100% reproducible. The best I've got is to add an Open SGID layer to the map, choose "Use Spatial Reference Extent", select and export and add to map, remove the Open SGID layer, add the Open SGID layer back but this time let it calculate the extent, and then try select and export. This maybe fails 50% of the time.

Talking with Michael, it seems like ArcMap does something to divine the spatial index grid settings for the output dataset (because PostGIS doesn't store the Esri spatial index like an SDE db would), and sometimes it gets it wrong and thus causes the export to fail (see https://support.esri.com/en/technical-article/000015090 and https://gis.stackexchange.com/questions/84317/exporting-data-from-oracle-to-esri-file-geodatabase-grid-size-error). Once I start getting the error, I've tried manually setting the grid size variables but it keeps giving me the error until I close/reopen the map. 

I've not been able to reproduce the error on ArcGIS Pro. It may handle the extent differently- after asking for the spatial extent of the Open SGID layer the first time I add it to a map, it doesn't ask for it again even if I remove the layer. 

@gis4ag , does it makes sense in your workflow to just export the entire county layer once and then work off of that locally?

**Comment by gis4ag** [comment]
Yes, I have sometimes exported the entire layer as a workaround. I think that would work in most cases. It's the randomness of the error that is frustrating. I'll keep a closer eye on the workflow at my end to see if I can find a clue as to what's happening.

Thanks so much for looking into this!

**Comment by jacobdadams** [comment]
Yeah, the randomness is definitely annoying. I'm sure there's something specific buried deep in ArcMap that triggers it, but given Esri's push towards Pro I'm not sure we'll have much luck getting to the bottom of it with them.
